lib/bluetooth_low_energy/aws_ble_gap.c

```diff
@@ -328,26 +331,32 @@ BTStatus_t BLE_ConnParameterUpdateRequest( const BTBdaddr_t * pxBdAddr,
 BTStatus_t BLE_ON( void )
 {
     /* Currently Disabled due to a bug with ESP32 : https://github.com/espressif/esp-idf/issues/2070 */
+
     /* xBTInterface.pxBTInterface->pxEnable(0); */
-	return eBTStatusSuccess;
+    return eBTStatusSuccess;
 }
 
+/*-----------------------------------------------------------*/
+
 BTStatus_t BLE_OFF( void )
 {
     BTStatus_t xStatus = eBTStatusSuccess;
-    Link_t  *pxConnectionListHead,  *pxConnectionListElem, *pxTemp;
+    IotLink_t * pxConnectionListHead, * pxConnectionListElem;
     BLEConnectionInfoListElement_t * pxConnInfo;
 
     xStatus = BLE_GetConnectionInfoList( &pxConnectionListHead );
+
     if( xStatus == eBTStatusSuccess )
     {
-        listFOR_EACH_SAFE( pxConnectionListElem, pxTemp, pxConnectionListHead )
+        /* Get the event associated to the callback */
+        IotContainers_ForEach( pxConnectionListHead, pxConnectionListElem )
```

Comment:
pxDisconnect() could trigger a dsiconnect() which can remove the connection from the list. Change to IotContainers_ForEachSafe ?

Reply:
Good point, I ll add it